comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -260,17 +262,22 @@ func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig,
 		return
 	}
 
+	modelConfig := pkgconfigsetup.Datadog()
+	if rc.isSystemProbe {
+		modelConfig = pkgconfigsetup.SystemProbe()
+	}
```

Comment:
Could we receive those as part of the dependencies instead of reaching for the global ?

Could we also use `pkg/util/flavor` to know if we're on system-probe instead of adding a boolean ?

Reply:
@hush-hush seems there is no system-probe in the flavor https://github.com/DataDog/datadog-agent/blob/main/pkg/util/flavor/flavor.go#L11